lib/Service/ProviderService.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar
+ * @copyright Anupam Kumar 2024
+ */
+
+namespace OCA\ContextChat\Service;
+
+use OCP\App\IAppManager;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\IUserManager;
+use Psr\Log\LoggerInterface;
+
+class ProviderService extends ProviderConfigService {
```

Comment:
Why are these two separate classes?

Reply:
One exclusively speaks to IConfig, the other also exposes functions to the controller so it was better to have them separate for logic separation